File: OBJCSample/OBJCSample/classes/AZPersianCalendar.m
//
//  AZPersianCalendar.m
//  AZCalendarOBJC
//
//  Created by Ali on 4/14/17.
//  Copyright © 2017 Ali Azadeh. All rights reserved.
//

#import "AZPersianCalendar.h"
#import "DateType.h"
@implementation AZPersianCalendar

+ (DateType*) getShamsiDate {
    DateType *result = [[DateType alloc] init];
    
    NSCalendar *gregorianCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
    
    
    
    
    NSDateComponents *gregorianComponents = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:[[NSDate alloc] initWithTimeIntervalSinceNow:0]];
    
   
    
    NSDate *date = [gregorianCalendar dateFromComponents:gregorianComponents];
    
    
    NSCalendar *persianCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierPersian];
    
    NSDateComponents *persianComponent = [persianCalendar components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:date];
    
    result.day = persianComponent.day;
    result.month = persianComponent.month;
    result.year = persianComponent.year;
    
    
    
    return  result;
}

+ (NSInteger) getPersianDayNumber{
    return [[AZPersianCalendar getShamsiDate] day];
}
+ (NSInteger) getPersianMonthNumber{
 return [[AZPersianCalendar getShamsiDate] month];
}
+ (NSInteger) getPersianYearNumber{
 return [[AZPersianCalendar getShamsiDate] year];
}

@end

File: SwiftSample/SwiftSample/Framework/AZCalendarSwift.framework/Headers/AZPersianCalendar.swift
//
//  AZPersianCalendar.swift
//  AZCalendarSwift
//
//  Created by Ali on 4/11/17.
//  Copyright © 2017 Ali Azadeh. All rights reserved.
//

import Foundation
public class AZPersianCalendar {
    
    
    
  public  static func getPersianDayNumber()->Int{
        
        return AZPersianCalendar.getPersianDate().day
    }
  public  static func getPersianMonthNumber()->Int{
        return AZPersianCalendar.getPersianDate().month
    }
   public static func getPersianYearNumber()->Int{
        return AZPersianCalendar.getPersianDate().year
    }
    
    
    private static func getPersianDate() -> (day : Int , month : Int , year : Int) {
        
        let gregorianCalendar = Calendar.init(identifier: .gregorian)
        let gregorianComponents = Calendar.current.dateComponents([.year, .month, .day], from: Date.init(timeIntervalSinceNow: 0))
        let date = gregorianCalendar.date(from: gregorianComponents)
        let persianCalendar = Calendar.init(identifier: .persian)
        let persianComponents = persianCalendar.dateComponents([.year, .month, .day], from: date!)
        return (day : persianComponents.day! , month : persianComponents.month! , year : persianComponents.year!)
          }
    
}
